diff --git a/src/compose.ts b/src/compose.ts
index b1d4508..76f84c3 100644
--- a/src/compose.ts
+++ b/src/compose.ts
@@ -30,7 +30,7 @@ export const compose = <E extends Env = Env>(
      * @returns {Promise<Context>} - A promise that resolves to the context.
      */
     async function dispatch(i: number): Promise<Context> {
-      if (i <= index) {
+      if (i < index) {
         throw new Error('next() called multiple times')
       }
       index = i
